### PR TITLE
feat(ECAM): add  fuel system alerts (low wing tk levels, CTR pumps off)

### LINKED ISSUE #4126: upper ecam fuel warnings and memos - ctr tanks and low fuel
**Is your feature request related to a problem? Please describe.**

Some upper ECAM warnings+memos are missing for the fuel system, below there are some examples mentioned.

**Describe the feature you'd like to see implemented**

Add ECAM warnings for fuel system events:
- CTR TK PUMPS OFF
- L(R) WING TK LO LVL
- L + R WING TK LO LVL
and corresponding memos.

**References**

LO WING TK level warning+memo is demonstrated in the official airbus Safety First | July 2015 edition (easy to find webpage).

**Additional context**

Discord username (if different from GitHub): Dunx#2505

### PR DESCRIPTION
**Pull request by Duncan10**
Fixes #4126

## Summary of Changes

Add to Upper ECAM warnings:
- CTR TK PUMPS OFF
- L(R) WING TK LO LVL
- L + R WING TK LO LVL

ADD ECAM memos for:
- CTR TK PUMPS OFF
- L(R) WING TK LO LVL
## Screenshots (if necessary)

grafik

grafik

grafik

## References

https://safetyfirst.airbus.com/fuel-monitoring-on-a320-family-aircraft/ :
grafik

## Additional context

Discord username (if different from GitHub): Dunx#2505

## Testing instructions

Verify memo for  CTR TK PUMPS OFF, and no memo for low level in both inner wing tanks.
Low level sensor in each tank triggered at 750kg threshold.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 1]
```diff
@@ -352,12 +352,47 @@ var A320_Neo_UpperECAM;
...
+                    message: "FUEL X FEED",
+                    action: "ON",
+                    isCompleted: () => this.getCachedSimVar("A:FUELSYSTEM VALVE SWITCH:3", "Bool"),
+                },
+                {
+                    style: "action",
+                    message: `${_tankSide} TK PUMP 1`,
+                    action: "OFF",
+                    isCompleted: () => {
+                        return (_tankSide == "R") ? !SimVar.GetSimVarValue("FUELSYSTEM PUMP SWITCH:3", "boolean") : !SimVar.GetSimVarValue("FUELSYSTEM PUMP SWITCH:2", "boolean");
```
Comment (tracernz) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:376]: Any reason not to use getCachedSimVar here and all the others below?

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 2]
```diff
@@ -1103,6 +1138,67 @@ var A320_Neo_UpperECAM;
...
+                                    const quantityR = SimVar.GetSimVarValue("FUEL TANK RIGHT MAIN QUANTITY", "gallons") * this.gallonToKg;
+                                    return (quantityL < 750) && (quantityR > 750);
+                                },
+                                actions: this.getLoWingTankActions("L")
+                            },
+                            {
+                                message: "CTR TK PUMPS OFF",
+                                level: 2,
+                                flightPhasesInhib: [1, 3, 4, 5, 7, 8, 9, 10],
+                                isActive: () => (!SimVar.GetSimVarValue("FUELSYSTEM PUMP SWITCH:1", "boolean") || !SimVar.GetSimVarValue("FUELSYSTEM PUMP SWITCH:4", "boolean")) && !SimVar.GetSimVarValue("L:A32NX_ECAM_INOP_SYS_FUEL PUMPS", "Bool") && SimVar.GetSimVarValue("FUEL TANK CENTER QUANTITY", "gallons") * this.gallonToKg > 250,
```
Comment (tracernz) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1171]: Maybe break this into multiple lines, it's veeery long.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 3]
```diff
@@ -1103,6 +1138,67 @@ var A320_Neo_UpperECAM;
...
+                                        message: "CTR TK PUMP 2",
+                                        action: "ON",
+                                        isCompleted: () => SimVar.GetSimVarValue("FUELSYSTEM PUMP SWITCH:4", "boolean"),
+                                    },
+
+                                ]
+
+                            },
+                            {
+                                message: "L + R WING TK LO LVL",
```
Comment (beheh) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1190]:
```suggestion
                                message: "L+R WING TK LO LVL",
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 4]
```diff
@@ -1103,6 +1138,67 @@ var A320_Neo_UpperECAM;
                             }
                         ]
 
+                    },
+                    {
+                        name: "Fuel",
```
Comment (beheh) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1143]:
```suggestion
                        name: "FUEL",
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 5]
```diff
@@ -352,12 +352,47 @@ var A320_Neo_UpperECAM;
                 }
             }
         }
+        getLoWingTankActions(_tankSide) {
+            return [
+                {
+                    style: "remark-indent-multiLine1",
+                    message: "IF NO FUEL LEAK",
```
Comment (beheh) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:359]:
```suggestion
                    message: "IF NO FUEL LEAK AND",
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 6]
```diff
@@ -352,12 +352,47 @@ var A320_Neo_UpperECAM;
...
+                },
+                {
+                    style: "remark-indent-multiLine2",
+                    message: "AND IMBALANCE",
```
Comment (beheh) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:363]:
```suggestion
                    message: "FUEL IMBALANCE",
```

**Review by beheh — CHANGES_REQUESTED** [review]
Accuracy pass.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js (hunk 1 above)
Comment (Duncan10) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: Thanks! adopted.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js (hunk 2 above)
Comment (Duncan10) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: changed. :)

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -233,7 +233,12 @@
 1. [MISC] Added custom loading tips - @Armankir (Arman#5297)
 1. [MCDU] Revised Reserve Weight Calculation and Default weight rule-set added - @Lucky38i (Lucky38i)
 1. [RMP] Added integral illumination to RMP volume knobs - @ImenesFBW (Imenes)
+<<<<<<< HEAD
```
Comment (beheh) [inline .github/CHANGELOG.md]: You've accidentally comitted the merge markers here.

Comment (Duncan10) [reply .github/CHANGELOG.md]: solved :)

**Review by beheh — APPROVED** [review]

**Comment by Matse08** [comment]
Category: QA Tester
Name: Matse#7454
Date of testing: 28.03.21
Version of the sim: Version the tester uses
PR Tested: #4126 
Tier of testing: 1
Changes to observe: New low fuel warnings


Testing technique: Try-out
Steps you've made to test the Pull Request:
1. Loaded in C&D
2. Turned on bat and ext pwr
3. Played with the fuel 



Media: 
Microsoft Flight Simulator 27 03 2021 23_14_57


Issues (If Any): 
Steps to achieve the issue

Overall Rating: Good
Conclusions of the testing: Nice one

**Review by tracernz — APPROVED** [review]
Nice work. 😃

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 29/03/2021
Version of the sim: 1.14.6.0
PR Tested: #41 
Tier of testing: 1
Changes to observe: 

**Steps you've made to test the Pull Request:**
1. Loaded onto RW with full fuel
2. Turned off CTR TK pumps, CTR TK PUMPS OFF ECAM comes on
3. Removed fuel to check LO LVL messages

**Media:**

**Issues (If Any):** 
- After crosschecking with the other Neo Pilots I have to say, that the neo does not have CTR TK pumps anymore. So the CTR TK PUMPS OFF ECAM is basically wrong in this airplane. Please try to gather some Neo fuel system reference in discord so we can get this accurate.
- Consequently your FUEL L+R WING TK LO LVL ECAM is also wrong for a neo, eg this is how this should look like

**Conclusions of the testing:** These messages are coded for the old CEO aircraft. Please consider recoding to NEO config after gathering appropriate reference. Contact us on discord so we can help you to get the right ones

**Comment by pilotJesse** [comment]
Category: QA Trial
Name: Jessee#8759
Date of testing: 01/04/2021
Version of the sim: 1.14.6.0
PR Tested: #5584
Tier of testing: 1
Changes to observe:
CTR TK PUMPS OFF
L(R) WING TK LO LVL
L + R WING TK LO LVL
CTR TK PUMPS OFF
L(R) WING TK LO LVL

Testing technique: 
1. Turned off CTR TK PUMPs off after 1st engine start and 1500ft AGL- recycled PB sw multiple times
2. Changed fuel in L, R & L+R to below 750kg (1650lb). 

Steps you've made to test the Pull Request:
1. Inserted full fuel, removed fuel L WING TK, R WING TK & L+R WINK TK
2. Turned off CTR TK FUEL PUMPS: after first engine start & above 1500ft.

Media: 
L+R WING TANK SC
L WING TANK SC
R WING TANK SC
CENTRE PUMPS OFF SC
CTR PUMPS OFF SC

Issues (If Any): 
No CTR TK PUMPS OFF ECAM displayed
Attempted test after 1st engine start and 1500ft AGL after inhibits.

Overall Rating: To be improved
Conclusions of the testing: FUEL LOW ECAMs working great, CENTRE TANK PUMPs off for me was not showing.

**Comment by St54Kevin** [comment]
@pilotJesse Thanks for the report. Next time please read at least the last comments/tests on the PR. Also this PR was marked as a Draft yesterday. So unfortunately your qa report was kind of useless, because its the wrong system :) 
@Duncan10 Please contact us on discord to discuss how to further proceed with this PR

**Comment by Benjozork** [comment]
Is this still being worked on ?

**Comment by MisterChocker** [comment]
Whats the status?

**Comment by Duncan10** [comment]
It's on hold until some depend on changes have been implemented e. g  fuel system, model with accurate overhead panel.

**Comment by aguther** [comment]
This PR is quite old, and a lot of things have changed in the meantime. I'm also not sure about the state overall. I'm closing it now, would be great to see a new PR when function is still not available.

**Pull request closed without merging**